ls1intum/Artemis
# `Assessment`: Fix the progress label in the submission dashboard for assessed exercises

### Checklist
#### General



#### Client


### Motivation and Context


**1. First problem:** During the assessment of the modeling exercises in the EIST course, I notice that the progress label is considering even the locked assessments to show the percentage of assessed submissions. This was confusing for the Feedback group because the progress label on the submission page was showing **"100% of the submissions assessed"**. However, this was wrong. A lock assessment cannot be considered an assessed submission. 
Screenshot 1:
<img width="1147" alt="image" src="https://user-images.githubusercontent.com/88991954/167013461-e75b8cbd-23bd-4c9b-be59-5cd2fc631e94.png">

The dashboard assessment was showing the correct number of assessed submissions. This should be also shown in the progress label. 
Screenshot 2:
![Screenshot 2022-05-04 at 21 27 09](https://user-images.githubusercontent.com/88991954/167013834-840fcbdb-7edc-46ee-ace1-0dd276afaaa9.png)

**2. Second problem:** This progress label was implemented (wrong) in the submissions of modeling exercises **only**. It was not implemented in other exercises that require manual assessment such as Text exercises, Programming exercises and Upload file exercises. 

**3. Third problem:** When there are no submissions, the page should not show the header of the table, and the progress label should point out that there are no submissions for that exercise. 
Screenshot 3:
<img width="1393" alt="image" src="https://user-images.githubusercontent.com/88991954/168776054-b5fe9b3f-b869-403e-a8be-1b30ebba49a9.png">




### Description


I have fixed the first problem. Now the locked submissions are not considered assessed submissions.

The following screenshot shows the assessment of 1/3 submission assessed.
Screenshot 4:
<img width="850" alt="image" src="https://user-images.githubusercontent.com/88991954/167311103-e59e370d-db6e-44f1-bd04-1dd81cd2743f.png">


The following screenshot shows a complete assessment (100%)
Screenshot 5:
<img width="889" alt="image" src="https://user-images.githubusercontent.com/88991954/167311020-96b6b28e-44e9-4450-92b3-8af572fe09c2.png">


I have implemented this progress label also in the submissions of text exercises, program exercises, and upload file exercises also. You can have this also in all kinds of exercises. 
Screenshot 6:
<img width="845" alt="image" src="https://user-images.githubusercontent.com/88991954/168777757-9f63e283-dded-4f89-9167-11cc241453d2.png">


It is also implemented that when there are no submissions, the header of the table is hidden and the progress label says that "No submissions for this exercise". 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 or more Students
- 1 Modeling, 1 text Exercise, 1 programming exercise, and 1 upload file exercise ( all of them with the manual assessment). 

1. Log in to Artemis (as instructor/tutor)
2. Navigate to Course Administration
3. Create 1 Modeling, 1 text Exercise, 1 programming exercise, and 1 upload file exercise ( all of them with the manual assessment). 
4. Check that without submissions, the header of the table is hidden (screenshot 6)
5. Log in as a student (student 1, student 2,  etc.). 
6. Resolve the exercises and submit your solution (as students 1, 2, etc.)
7. Log in as an instructor or tutor 
8. Go to submissions of the text, modeling, programming, and upload file exercises. 
**Let's assume you have 2 submissions**
9. Without any assessments, the progress label should show 0% (0 out of 2 submitted models assessed). 
11. Assess one exercise and **submit your assessment**. Verify that the progress label shows 50% (1 out of 2 submitted **models/text/program/upload** file exercise assessed). This must be in orange colour. **Check also in dark mode**
12. Assess the second exercise and **DO NOT SUBMIT YOUR ASSESSMENT** Verify that the progress label shows 50% (1 out of 2 submitted models assessed). This must be in orange color again. This is because you just locked the assessment. This can be verified by clicking on the **Show locked option**. 
13. Get in again in the second submission and **submit your assessment**.  Verify that the progress label shows 100% (2 out of 2 submitted models assessed). This must be in green color. Because all submissions have been assessed. **Check also in dark mode.**
14. Check that the progress label works also in German. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots


The following screenshots show how the progress label looks in dark mode
<img width="781" alt="image" src="https://user-images.githubusercontent.com/88991954/167714973-dee4542f-a405-45ea-8964-7f8ccdb138f9.png">

<img width="764" alt="image" src="https://user-images.githubusercontent.com/88991954/167715179-74f4f929-4f25-46e9-8997-cbb966d425bb.png">
